temporal-sdk/src/main/java/io/temporal/worker/WorkflowImplementationOptions.java

```diff
@@ -64,36 +68,51 @@ public Builder setFailWorkflowExceptionTypes(
       return this;
     }
 
+    public Builder setActivityOptions(Map<String, ActivityOptions> activityOptions) {
+      this.activityOptions = activityOptions;
+      return this;
+    }
```

Comment:
I would also add ability to specify default activity options like:
```
setDefaultActivityOptions(ActivityOptions activityOptions) 
```
which would have the lowest precedence across all the options.